indexd/drs/blueprint.py

```diff
@@ -248,17 +248,19 @@ def bundle_to_drs(record, expand=False, is_content=False):
 
 def parse_checksums(record, drs_object):
     ret_checksum = []
-    if "hashes" in record:
-        for k in record["hashes"]:
-            ret_checksum.append({"checksum": record["hashes"][k], "type": k})
-    elif "checksum" in record:
-        checksums = json.loads(record["checksum"])
-        for checksum in checksums:
-            ret_checksum.append(
-                {"checksum": checksum["checksum"], "type": checksum["type"]}
-            )
-
-    return ret_checksum
+    try:
+        if "hashes" in record:
+            for k in record["hashes"]:
+                ret_checksum.append({"checksum": record["hashes"][k], "type": k})
+        elif "checksum" in record:
+            checksums = json.loads(record["checksum"])
+            for checksum in checksums:
+                ret_checksum.append(
+                    {"checksum": checksum["checksum"], "type": checksum["type"]}
+                )
+        return ret_checksum
+    except KeyError:
+        raise UserError("Object does not contain checksum")
```

Comment:
if the object doesn't contains `hashes` or `checksum`, you are not throwing an error but not returning anything. Since the checksums are not optional, you need to throw an error. https://github.com/uc-cdis/indexd/compare/chore/parse_checksums#diff-926a6ca016e96ef2492a0f1d27a5c220339f49274d0caea713c0403706869d34R260